args/core/async/transferable_atomic.hpp

```diff
@@ -1,21 +1,34 @@
 #pragma once
 #include <atomic>
+#include <core/async/readonly_rw_spinlock.hpp>
+
+/** @file transferable_atomic.hpp
+ */
 
 namespace args::core::async
 {
+	/**@class transferable_atomic
+	 * @brief Copyable wrapper for std::atomic.
+	 */
 	template <typename T>
 	struct transferable_atomic
 	{
 	private:
 		std::atomic<T> m_atomic;
+		mutable readonly_rw_spinlock m_lock;
 	public:
 		transferable_atomic() noexcept = default;
 
-		constexpr transferable_atomic(T val) noexcept : m_atomic(val) {}
+		constexpr transferable_atomic(T val) noexcept : m_atomic(val), m_lock() {}
+
+		transferable_atomic(const std::atomic<T>& other) : m_atomic(other.load(std::memory_order_acquire)), m_lock() {}
 
-		transferable_atomic(const std::atomic<T>& other) : m_atomic(other.load(std::memory_order_acquire)) {}
+		transferable_atomic(const transferable_atomic<T>& other) : m_atomic(other->load(std::memory_order_acquire)), m_lock() {}
```

Comment:
considering that it implements a copy-ctor, and the type is quite literally called `transferable_atomic` you might want to also implement a move-constructor, a copy assignment operator, a move assignment operator and a destructor.
(see cppcoreguidelines-special-member-functions)
when these do not apply, explicitly mark them as `= delete;`